### ISSUE TITLE
Add Hetzner provider

### ISSUE THREAD

**Issue by sa7mon** [sa7mon/S3Scanner#378]
https://docs.hetzner.com/storage/object-storage/overview

**Comment by LefterisXefteris** [comment]
Hello mate, can i contribute ? it's really exciting what are you doing!!!!

**Comment by sa7mon** [comment]
> Hello mate, can i contribute ? it's really exciting what are you doing!!!!

Absolutely! What you'll need to do is create a `provider/hetzner.go` that fulfills the StorageProvider interface. It would also be helpful to find a couple of buckets to use for testing that have different permissions set.

The last thing to add is a region check at `cmd/regioncheck/hetzner.go` that returns a list of supported regions. This is so we can periodically make sure that every region possible is being checked when scanning. Feel free to use the DigitalOcean or Wasabi region checks as an example.

**Comment by LefterisXefteris** [comment]
got it, i am working on it!

**Comment by LefterisXefteris** [comment]
Hello there,

I am using the AWS CLI to connect Hetzner Cloud to my local machine and to test whether a bucket exists in Hetzner Cloud.

When I build and run it with the bucket name ‘S3Scanner,’ it automatically detects the bucket without implementing the hetzner.go file.

**Comment by sa7mon** [comment]
> Hello there,
> 
> I am using the AWS CLI to connect Hetzner Cloud to my local machine and to test whether a bucket exists in Hetzner Cloud.
> 
> When I build and run it with the bucket name ‘S3Scanner,’ it automatically detects the bucket without implementing the hetzner.go file.

Can you provide the full command you ran? if you only give the `-bucket` argument, the default provider is AWS.

**Comment by LefterisXefteris** [comment]
Hello there sorry for the late reply, had some personal issues.

i ran this lefterisgilmaz@MacBookPro S3Scanner % ./s3scanner -bucket s3scanner                
INFO exists    | s3scanner | us-west-1 | AuthUsers: [] | AllUsers: [] . Hetzner creds are stores in the default aws cli.

**Comment by sa7mon** [comment]
Yep, if you don't provide the `-provider` flag, it will default to AWS. I indeed have several S3 buckets in AWS named permutations of `s3scanner` for testing.
